domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -259,9 +261,19 @@ constructor(
         val eligibleContentValues = getEligibleContentValues()
         val outputFileOptions: OutputFileOptions
         if (imageCaptureUri == null) {
-            val e = RuntimeException("Null Uri is provided.")
-            Log.d(TAG, "takePicture onError: $e")
-            throw e
```

Comment:
Since this no longer throws an exception on null, we need to add a null check for the intent flow outside of this method.

Reply:
Can we just make this the default behavior? If null url is provided, we just save it  to the default location. I added a javaDoc on CameraUseCase.takePicture() for this behavior

Reply:
It's fine for that to be the behavior for CameraUseCase.takePicture(), but we still need to validate the URI isn't null when we are handling `ACTION_IMAGE_CAPTURE`. Technically we should be returning a small Bitmap in that case according to https://developer.android.com/reference/android/provider/MediaStore#ACTION_IMAGE_CAPTURE, but since we don't plan on doing that currently, we should instead show an error. We don't want to have the side-effect of saving a picture that wasn't requested.